src/main/java/ui/MenuControl.java

```diff
@@ -317,6 +325,59 @@ private MenuItem createRefreshMenuItem() {
         return new MenuItem[] { newIssueMenuItem, newLabelMenuItem, newMilestoneMenuItem };
     }
 
+    private MenuItem[] createReposMenu() {
+        Menu remove = new Menu("Remove");
+
+        ui.registerEvent((OpenReposChangedEventHandler) e -> {
+            Platform.runLater(() -> updateRepoRemoveList(remove));
+        });
+
+        return new MenuItem[] { remove };
+    }
+
+    private void updateRepoRemoveList(Menu remove) {
+        remove.getItems().clear();
+
+        HashSet<String> currentlyUsedRepo = getCurrentlyUsedRepo();
+
+        for (String repoId : ui.logic.getStoredRepos()) {
+            if (!currentlyUsedRepo.contains(repoId)) {
+                MenuItem removeItem = new MenuItem(repoId);
+                removeItem.setOnAction(e1 -> onRepoRemove(repoId));
+                remove.getItems().add(removeItem);
+            }
+        }
+    }
+
+    private HashSet<String> getCurrentlyUsedRepo() {
+        HashSet<String> currentlyUsedRepos = new HashSet<String>();
```

Comment:
Yes, I meant referenced by some filter. I can't use
`logic.getOpenRepositories()` since repo is only so-called closed when
HubTurbo quits. I guess it's for caching reason. By open, it means there is
a model for the repo (deleting the json would not close the repo either).
Right now there is no such close method.
On Sep 7, 2015 7:01 PM, "Darius Foo" <notifications@github.com> wrote:

> In src/main/java/ui/MenuControl.java
> <https://github.com/HubTurbo/HubTurbo/pull/940#discussion_r38853149>:
>
> > +    private void updateRepoRemoveList(Menu remove) {
> > +        remove.getItems().clear();
> > +
> > +        HashSet<String> currentlyUsedRepo = getCurrentlyUsedRepo();
> > +
> > +        for (String repoId : ui.logic.getStoredRepos()) {
> > +            if (!currentlyUsedRepo.contains(repoId)) {
> > +                MenuItem removeItem = new MenuItem(repoId);
> > +                removeItem.setOnAction(e1 -> onRepoRemove(repoId));
> > +                remove.getItems().add(removeItem);
> > +            }
> > +        }
> > +    }
> > +
> > +    private HashSet<String> getCurrentlyUsedRepo() {
> > +        HashSet<String> currentlyUsedRepos = new HashSet<String>();
>
> Okay, sure. I guess by 'in use' you mean 'referenced from some filter'?
>
> My original question was whether you could use logic.getOpenRepositories(),
> because it makes sense that you should be able to close any repo that is
> currently open, even if it isn't referenced from some filter.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/HubTurbo/HubTurbo/pull/940/files#r38853149>.
>